gina-alaska/viirs-snow-metrics
# Implement water masking

This PR accomplishes a few things:
 - Modifies directory configuration such that `SNOW_YEAR` subdirectories are generated and used.
 - Implements a masking script. This script creates three mask files. One for ocean, one for lake / inland water, and one for all water (a combination of ocean and lake / inland water).
 - Updates the README to better reflect a practical configuration for Elephant. Some minor usage, disk space, and execution time updates are also included.


To test this PR:
 - Go ahead and wipe your pre-existing downloaded data and preprocessed datasets.
 - Read the README - it has been updated! Set your environment variables accordingly. Pick a random snow year though. Something between 2013 and 2021 *should* work. I've been using 2015, but we may as well test other years.
 - Active your `viirs-snow` conda env.
 - Run `python download.py` and then go work on something else or take a walk, this might take all day. Give the log file a quick scan. Examine the download: `ls -lhrt $INPUT_DIR/$SNOW_YEAR` or similar.
 - Run `python preprocess.py $tile_id` and then examine the output directory and scan the log file. Maybe try `h10v02` or `h12v02`. I've been testing `h11v02`.
 - Run `python watermask.py $tile_id` adn then examine the output directory and scan the log file.
 - Examine the mask GeoTIFFs in QGIS or ArcGIS. Bring them to your local machine with something like `scp -r "$USER"@elephant.gina.alaska.edu://export/datadir/$USER"_viirs_snow/$SCRATCH_DIR/$SNOW_YEAR/masked .` Check out data values and visual integrity. Do they pass the eyeball test? Compare them to a map service or to a few of the downloaded source GeoTIFFs.

Closes #11
